doc/book/validators/undisclosed-password.md

```diff
@@ -0,0 +1,38 @@
+# Undisclosed Password Validator
+
+`Zend\Validator\UndisclosedPassword` allows you to validate if a given password was found in data breaches using the service [Have I Been Pwned?](https://www.haveibeenpwned.com), in a secure, anonymous way using [K-Anonymity](https://www.troyhunt.com/ive-just-launched-pwned-passwords-version-2) to ensure passwords are not send in full over the wire.
+
+> ## Installation requirements
+> 
+> This validator needs to make a request over HTTP, therefor it requires an HTTP client of your choice that implements [PSR-18](https://www.php-fig.org/psr/psr-18/) and [PSR-17](https://www.php-fig.org/psr/psr-17/) request and response factories.
+>
+> Make sure you have it installed before using this validator:
+>
+> ```bash
+> $ composer require psr/http-client
+> $ composer require psr/http-factory
+> ```
+
+## Basic usage
+
+To validate if a password was disclosed in a known data breach, you need to provide a HTTP Client that implements `psr\http\ClientInterface`, a `Psr\Http\Message\RequestFactoryInterface` and a `Psr\Http\Message\ResponseFactoryInterface` to the constructor and validate the password you want to check.
+
+If the password was found via the service, `isValid` will return FALSE. If the password was not found, `isValid` will return TRUE.
+
+```php
+$validator = new Zend\Validator\UndisclosedPassword(
```

Comment:
The example must be completely reproducible for the end user. At the moment, the example shows the same thing as the API, which is not very helpful. We need a full code example including a HTTP client and the factories.

Reply:
I will update this piece of example code after I figure out which PSR-18 http client I could use in this example. 

I was thinking about using the Guzzle6 adapter and factories, because zend-diactoros doesn't implement PSR-18 (yet), but I can go any direction.

Reply:
@froschdesign In commit 1383af3 I have provided a working example of `Zend\Validator\UndisclosedPassword` using PSR7, PSR17 and PSR18 HTTP Client and Messages.

Please have a look at it if this is in line with your expectations. If not, please indicate how I can improve it.